components/hydrator/internal/tenantmapping/access_level_context_provider.go

```diff
@@ -86,33 +97,30 @@ func (p *accessLevelContextProvider) Match(_ context.Context, data oathkeeper.Re
 	}
 
 	if _, err := data.GetExternalTenantID(); err != nil {
-		if apperrors.IsKeyDoesNotExist(err) {
-			return false, nil, nil
+		if !apperrors.IsKeyDoesNotExist(err) {
+			return false, nil, err
 		}
-		return false, nil, err
 	}
 
 	return true, &oathkeeper.AuthDetails{AuthID: idVal, AuthFlow: oathkeeper.CertificateFlow, CertIssuer: certIssuer}, nil
 }
 
-func (p *accessLevelContextProvider) verifyTenantAccessLevels(tenantType string, authDetails oathkeeper.AuthDetails, reqData oathkeeper.ReqData) error {
+func (p *accessLevelContextProvider) verifyTenantAccessLevels(accessLevel string, authDetails oathkeeper.AuthDetails, reqData oathkeeper.ReqData) error {
```

Comment:
The logic before was the same, but instead of just returning and stopping the function some boolean flag and if / else's were used. This is just a styling change. The header is checked in the GetObjectContext where the business logic of that object context provider resides.

Reply:
I see, I've missed the `reqData.GetExternalTenantID()` call in the `GetObjectContext` method.
In such case do we need the `reqData.GetExternalTenantID ` invocation in the `verifyTenantAccessLevels` method? as we already check that before invoke the "verify" method

Reply:
In `verifyTenantAccessLevels` we don't call it to verify something, we call it to have more meaningful information to the error message.